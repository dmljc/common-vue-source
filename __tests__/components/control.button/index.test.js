/**
 * @jest-environment jsdom
 */
import { createLocalVue, mount } from '@vue/test-utils';
import Compose from '@/main';
import ControlButton from '@/components/control.button';

let customMount;
beforeAll(() => {
    const localVue = createLocalVue();
    localVue.use(Compose);
    customMount = options => mount(ControlButton, Object.assign({localVue}, options));
});

describe('control-button base usage', () => {
    it('control-button matchs snapshot case1', () => {
        const wrapper = customMount({
            slots: {
                default: '导出'
            }
        });
        expect(wrapper.props('type')).toBe('default');
        expect(wrapper.props('size')).toBe(undefined);
        expect(wrapper.props('icon')).toBe('');
        expect(wrapper.props('nativeType')).toBe('button');
        expect(wrapper.props('disabled')).toBe(false);
        expect(wrapper.props('plain')).toBe(false);
        expect(wrapper.props('autofocus')).toBe(false);
        expect(wrapper.vm.loading).toBe(false);
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('control-button advanced usage', () => {
    it('control-button matchs snapshot case2', () => {
        const wrapper = customMount({
            slots: {
                default: '导出'
            },
            propsData: {
                type: 'text',
                size: 'small',
                icon: 'el-icon-info',
                plain: true,
                autofocus: true,
                nativeType: 'submit'
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('control-button loading function', () => {
    it('control-button loading function', (done) => {
        const wrapper = customMount({
            slots: {
                default: '导出'
            },
            listeners: {
                click: (e) => {
                    expect(wrapper.vm.loading).toBe(true);
                    e.resolve();
                    expect(wrapper.vm.loading).toBe(false);
                    done();
                }
            }
        });
        wrapper.trigger('click');
    });
});
