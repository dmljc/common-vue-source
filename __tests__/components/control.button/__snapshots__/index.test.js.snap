// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control-button advanced usage control-button matchs snapshot case2 1`] = `
<button
  autofocus="autofocus"
  class="el-button el-button--text el-button--small is-plain"
  type="submit"
>
  <!---->
  <i
    class="el-icon-info"
  />
  <span>
    导出
  </span>
</button>
`;

exports[`control-button base usage control-button matchs snapshot case1 1`] = `
<button
  class="el-button el-button--default"
  type="button"
>
  <!---->
  <!---->
  <span>
    导出
  </span>
</button>
`;
